atomic_reactor/plugins/pre_inject_yum_repos.py

```diff
@@ -0,0 +1,264 @@
+from urllib.parse import urlparse
+
+from atomic_reactor.config import get_koji_session
+from atomic_reactor.constants import YUM_REPOS_DIR, RELATIVE_REPOS_PATH, INSPECT_CONFIG, \
+    PLUGIN_RESOLVE_COMPOSES_KEY
+from atomic_reactor.dirs import BuildDir
+from atomic_reactor.plugin import PreBuildPlugin
+from atomic_reactor.util import is_scratch_build, map_to_user_params, \
+    allow_repo_dir_in_dockerignore, render_yum_repo
+from atomic_reactor.utils.yum import YumRepo
+
+
+class InjectYumReposPlugin(PreBuildPlugin):
+    key = "inject_yum_repos"
+    is_allowed_to_fail = False
+
+    args_from_user_params = map_to_user_params(
+        "target:koji_target",
+        "repourls:yum_repourls",
+    )
+
+    def validate_yum_repo_files_url(self):
+        if not self.allowed_domains:
+            return
+        errors = []
+
+        for repourl in self.repourls:
+            repo_domain = urlparse(repourl).netloc
+            if repo_domain not in self.allowed_domains:
+                errors.append('Yum repo URL {} is not in list of allowed domains: {}'
+                              .format(repourl, self.allowed_domains))
+
+        if errors:
+            raise ValueError('Errors found while checking yum repo urls: \n{}'
+                             .format('\n'.join(errors)))
+
+    def _final_user_line(self):
+        user = self._find_final_user()
+        if user:
+            return user
+
+        if not self.workflow.dockerfile_images.base_from_scratch:
+            # Inspect any platform: the User should be equal for all platforms
+            inspect = self.workflow.imageutil.base_image_inspect()
+            user = inspect.get(INSPECT_CONFIG, {}).get('User')
+            if user:
+                return f'USER {user}'
+
+        return ''
+
+    def _find_final_user(self):
+        """Find the user in USER instruction in the last build stage"""
+        for insndesc in reversed(self._dockerfile.structure):
+            if insndesc['instruction'] == 'USER':
+                return insndesc['content']  # we will reuse the line verbatim
+            if insndesc['instruction'] == 'FROM':
+                break  # no USER specified in final stage
+
+    def _cleanup_lines(self):
+        lines = [
+            "RUN rm -f " + " ".join(
+                (f"'{repo.dst_filename}'" for repo in self.yum_repos)
+            )
+        ]
+        if self._builder_ca_bundle:
+            lines.append(f'RUN rm -f /tmp/{self._ca_bundle_pem}')
+
+        final_user_line = self._final_user_line()
+        if final_user_line:
+            lines.insert(0, "USER root")
+            lines.append(final_user_line)
+
+        return lines
+
+    def add_koji_repo(self):
+        self.xmlrpc = get_koji_session(self.workflow.conf)
+        self.pathinfo = self.workflow.conf.koji_path_info
+        self.proxy = self.workflow.conf.yum_proxy
+
+        if not self.target:
+            self.log.info('no target provided, skipping plugin')
```

Comment:
```suggestion
            self.log.info('no target provided, not adding koji repo')
```
(the rest of the plugin is not skipped)

Reply:
added this change